pkg/placement/what-resolver.go

```diff
@@ -512,6 +516,21 @@ func (wr *whatResolver) processEdgePlacement(ctx context.Context, cluster logica
 	epFound := err == nil
 	wr.Lock()
 	defer wr.Unlock()
+
+	//change to consumer SpaceID
+	_, epOriginalName, kbSpaceID, err := kbuser.AnalyzeObjectID(ep)
+	if err != nil {
+		logger.Error(err, "object does not appear to be a provider's copy of a consumer's object")
+		return true
+	}
+	spaceID := wr.kbSpaceRelation.SpaceIDFromKubeBind(kbSpaceID)
+	if spaceID == "" {
+		logger.Error(nil, "failed to get consumer space ID from a provider's copy")
+		return true
+	}
+	cluster := logicalcluster.Name(spaceID)
+	epName = ObjectName(epOriginalName)
```

Comment:
It is not necessary to hold the mutex while doing this stuff, right?

Reply:
Absolutely. Changed